src/oneD/Sim1D.cpp

```diff
@@ -424,6 +429,9 @@ void Sim1D::solve(int loglevel, bool refine_grid)
             writeline('.', 78, true, true);
         }
         while (!ok) {
+             // Keep the attempt_counter in the range of [1, max_history]
+             attempt_counter = (attempt_counter % max_history) + 1;
+
```

Comment:
I like this idea of keeping a somewhat longer history in the `debug_sim1d` file. I have a couple of suggestions for refinement:
- You might consider setting the name to just `solution_N`, as in the case where the "type" of the state changes, you can end up with more than `max_history` entries in the file, for example, I got a file with keys: `['solution_1_NewtonFail', 'solution_1_NewtonSuccess', 'solution_1_Regridding', 'solution_1_Timestepping', ...]`
- To help with understanding the order once you exceed `max_history`, I think it would be better to add the full value (non modulo) to the description, and only take the modulus for the purpose of setting the "name".

Reply:
I did notice that when I call solve() and then call solve() again, if the second fails, it makes for a confusing debug file because we get that success and fail entry present for the solution_1. At the time I was torn between keeping that behavior as we probably only should be examining a failure within a call to solve().

Reply:
Would it make sense to delete `debug_sim1d.yaml` before the first write? At which point, there's probably a small bit of refactoring to move the repetitive debug blocks in `Sim1D::solve` into their own function.

Reply:
I did that. In the other PR, I'll push the changes shortly.